nxdrive/engine/dao/sqlite.py

```diff
@@ -752,6 +752,22 @@ def _migrate_db(self, cursor: Cursor, version: int) -> None:
 
             self.store_int(SCHEMA_VERSION, 9)
 
+        if version < 10:
+            # Remove States with bad digests.
+            # Remove Downloads linked to these States.
+
+            for doc_pair in cursor.execute(
+                "SELECT * FROM States WHERE remote_digest IS NOT NULL;"
+            ):
+                digest = doc_pair["remote_digest"]
+                if digest.count('"') != 0 or not get_digest_algorithm(digest):
```

Comment:
```suggestion
                if not get_digest_algorithm(digest):
```

Think to add a debug log line when removing an entry.